src/content/blog/2024-08-05-rafiki-code-architecture.mdx

```diff
@@ -0,0 +1,203 @@
+
+### The Rafiki Authorization Server
+
+The Authorization Server on Rafiki is structured similarly to the backend, in that it injects services into an IoC container which are retrieved by routes on Koa server to perform the business logic.
+
+[index.ts](https://github.com/interledger/rafiki/blob/main/packages/auth/src/index.ts)
+```
+import { Ioc, IocContract } from '@adonisjs/fold'
+import { createGrantRoutes } from './grant/routes'
+
+...
+
+const container: IocContract<AppServices> = new Ioc()
+
+...
+
+container.singleton('grantRoutes', async (deps: IocContract<AppServices>) => {
+  return createGrantRoutes({
+    grantService: await deps.use('grantService'),
+    clientService: await deps.use('clientService'),
+    accessTokenService: await deps.use('accessTokenService'),
+    accessService: await deps.use('accessService'),
+    interactionService: await deps.use('interactionService'),
+    logger: await deps.use('logger'),
+    config: await deps.use('config')
+  })
+})
+```
+
+Again, note the route service for grants is mounted to the container, and then subsequently referenced when attaching service functions to the relevant routes.
+
+[app.ts](https://github.com/interledger/rafiki/blob/main/packages/auth/src/app.ts)
+```
+const grantRoutes = await this.container.use('grantRoutes')
+
+...
+
+/* Back-channel GNAP Routes */
+// Grant Initiation
+router.post<DefaultState, CreateContext>(
+  '/',
+  createValidatorMiddleware<CreateContext>(openApi.authServerSpec, {
+    path: '/',
+    method: HttpMethod.POST
+  }),
+  grantInitiationHttpsigMiddleware,
+  grantRoutes.create
+)
+```
+
+Like the `backend`, the `auth` server also extends a GraphQL API to perform admin functions. The API is modeled by this [GraphQL schema](https://github.com/interledger/rafiki/blob/main/packages/auth/src/graphql/schema.graphql).
+
+### The Rafiki Frontend
+
+The Rafiki has a React-based (specifically [Remix](https://remix.run/docs/en/main)) frontend with server-side rendering that consumes the GraphQL API extended by the `backend` server.
+In general, the [name of a file](https://github.com/interledger/rafiki/tree/main/packages/frontend/app/routes) dictates how the app constructs its routes. For example, the file [assets.create.tsx](https://github.com/interledger/rafiki/blob/main/packages/frontend/app/routes/assets.create.tsx) is expressed as `/assets/create` in the frontend app. Path parameters are denoted by a `$` sign, so the file [assets.$assetId.tsx](https://github.com/interledger/rafiki/blob/main/packages/frontend/app/routes/assets.%24assetId.tsx) might be expressed as `/assets/1234-1234-12345678`, where the `$assetId` portion of the file name stands in for an identifier in the final route.
+
+<LargeImg src="/developers/img/blog/2024-08-05/rafiki-admin-screen.png" alt="A screenshot of the Rafiki Admin UI" />
```

Comment:
```suggestion
![A screenshot of the Rafiki Admin UI](/developers/img/blog/2024-08-05/rafiki-admin-screen.png)
```
This doesn't require the ability to view it in a separate screen for small details to be more visible.